### PR TITLE
Merge future-stabilization into master

### PR DESCRIPTION
**Pull request by vslsnap**
This is an automatically generated pull request from future-stabilization into master.

@dotnet/roslyn-infrastructure:

``` bash
git remote add vslsnap-roslyn "https://github.com/vslsnap/roslyn.git"
git fetch vslsnap-roslyn
git fetch upstream
git checkout merge-future-stabilization-into-master20160621-150044
git reset --hard upstream/master
git merge upstream/future-stabilization
# Fix merge conflicts
git commit
git push vslsnap-roslyn merge-future-stabilization-into-master20160621-150044 --force
```

Once the merge can be made and all the tests pass, you are free to merge the pull request.

### CODE REVIEW

**Comment by vslsnap** [comment]
@dotnet-bot test vsi please

File: src/Compilers/CSharp/Portable/Errors/MessageID.cs [hunk 1]
```diff
@@ -164,6 +164,30 @@ internal static string RequiredFeature(this MessageID feature)
             return null;
         }
 
+        internal static string RequiredFeatureLegacy(this MessageID feature)
```
Comment (jinujoseph) [inline src/Compilers/CSharp/Portable/Errors/MessageID.cs]: @jaredpar , can you pls review the changes to this file.

**Comment by jinujoseph** [comment]
Thanks @agocke for helping with conflict resolution

File: src/Compilers/CSharp/Portable/Errors/MessageID.cs (hunk 1 above)
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/Errors/MessageID.cs]: We should delete this function entirely.

**Comment by jinujoseph** [comment]
@dotnet-bot test windows_debug_unit64_prtest please

**Comment by jinujoseph** [comment]
@dotnet-bot retest windows_debug_unit64_prtest please

**Pull request merged**